# Week 8 Mini Assignment

In [1]:
import pandas as pd

comments = pd.read_excel("C4Comments.xlsx")
courses = pd.read_excel("C4Courses.xlsx")

## Comments data

In [8]:
comments.head()

Comment_ID  ...                                            Comment
0     7197674  ...  This is a good question, i do think it's a lit...
1     7197679  ...  This is a really good question, i like the str...
2     7199285  ...  It is a good question. Slightly tricky but not...
3     7202419  ...  I think this is a really good question! I thin...
4     7202425  ...  This question requires the students to recall ...

[5 rows x 6 columns]

In [39]:
print("There are {} rows in the comments data source".format(comments.shape[0]))
print("There are {} unique comment_ids".format(comments['Comment_ID'].nunique()))

There are 2229 rows in the comments data source
There are 2229 unique comment_ids


#### Comments overview
- We have 2229 rows of data
- Each row is a unique comment for a particular course
- Comment column contains the actual comment made
- Also have access to the userID of the user who made the comment and the question id of where the comment was 

#### How many different students are represented by this data?

In [21]:
print("There are {} unique students in the comments data".format(comments['user'].nunique()))

There are 269 unique students in the comments data


### How many different questions are represented by this data?

In [25]:
print("There are {} unique question_ids".format(comments['Question_ID'].nunique()))

There are 441 unique question_ids


#### What is the date range of the data?

In [38]:
oldest = pd.to_datetime(comments['Timestamp'].min())
newest = pd.to_datetime(comments['Timestamp'].max())
print("The oldest date is {} and the newest date is {}".format(oldest, newest))
print("This is a date range of {}".format((newest-oldest)))

The oldest date is 2020-04-28 21:23:10 and the newest date is 2020-05-19 12:20:26
This is a date range of 20 days 14:57:16


## Now let's look at the Courses data

In [40]:
courses.head()

CourseID      CourseName  StartDate    Deadline_Comment   Deadline_Question
0     21320    NUTM2001 Mon 2020-04-27 2020-05-12 01:00:10 2020-05-05 01:00:10
1     21326   NUTM2001 Tues 2020-04-28 2020-05-13 01:00:10 2020-05-06 01:00:10
2     21327    NUTM2001 Wed 2020-04-29 2020-05-14 01:00:10 2020-05-07 01:00:10
3     21328  NUTM2001 Thurs 2020-04-30 2020-05-15 01:00:10 2020-05-08 01:00:10

- Contains course names as well as the day
- Contains course ID column which is also reflected by the coursename column
- Contains deadline information for comments and questions

### How many rows are there?

In [44]:
print("There are {} rows of data".format(courses.shape[0]))

There are 4 rows of data


- Only 4 rows so only 4 courses
- This data source is probably only useful for the deadline columns but even then doesn't seem too useful at an initial glance